HACKING.md

```diff
@@ -198,6 +198,33 @@ tox -e docs
 open docs/_build/html/index.html
 ```
 
+## How to Pull in Style Changes
+
+The documentation uses Canonical styling which is customised on top of the [Furo Sphinx theme](https://github.com/pradyunsg/furo). The easiest way to pull in Canonical style changes is by using the Canonical documentation starter pack, see docs [here](https://canonical-starter-pack.readthedocs-hosted.com/) and repository [here](https://github.com/canonical/sphinx-docs-starter-pack).
+
+TL;DR:
+
+- Clone the starter pack repository to a local directory: `git clone git@github.com:canonical/sphinx-docs-starter-pack`.
+- Copy the folder `.sphinx` under the starter pack repo to the operator repo `docs/.sphinx`.
+
+## How to Customise Configurations
+
+There are two configuration files: [`docs/conf.py`](./docs/conf.py) and [`docs/custom_conf.py`](./docs/custom_conf.py), copied and customised from the starter pack repo.
+
+To customise, change the file [`docs/custom_conf.py`](./docs/custom_conf.py) only, and theoretically, we should not change [`docs/conf.py`](./docs/conf.py) (however, some changes are made to [`docs/conf.py`](./docs/conf.py), such as adding autodoc, PATH, fixing issues, etc.)
+
+## How to Pull in Dependency Changes
+
+The Canonical documentation starter pack uses Make to build the documentation, which will run the script [`docs/.sphinx/build_requirements.py`](./docs/.sphinx/build_requirements.py) and generate a requirement file `requirements.txt` under `docs/.sphinx/`.
```

Comment:
We don't seem to have a Makefile, and `make html` (documented below) doesn't work. Should it?

Reply:
`make html` should run within the starter pack repo, not our repo.

The starter pack uses Makefile to run the `build_requirements.py` script which generates a `requirements.txt` dynamically, then build the docs based on that.

Since the OF repo uses `pip-compile` to create a requirements based on both `pyproject.toml` and a `requirements.txt` for the docs, I did not include the Makefile into the OF repo, just the `requirements.txt` generated by the `Makefile`/`build_requiremenets.py`. Come to think of this, maybe I should remove the `build_requirements.py` as well, or include the `Makefile`?

Reply:
Per voice discussion, let's use tox for the pip-compile step, and keep the initial build_requirements.py out of this repo altogether (you just run that once when you're copying the starter pack).

Reply:
After discussion we decide to not include build_requiremenets.py at all, only the generated requirements.txt.